### PR TITLE
The steps to set-up Python-RtMidi in Windows Bash

### PR DESCRIPTION
**Pull request by Fleker**

### CODE REVIEW

File: INSTALL.rst.in [hunk 1]
```diff
@@ -175,3 +175,36 @@ Please see the detailed instructions for Windows in the file
 .. _jack for os x: http://www.jackosx.com/
 .. _pyliblo: http://das.nasophon.de/pyliblo/
 .. _pyyaml: https://pypi.python.org/pypi/PyYAML
+
```
Comment (SpotlightKid) [inline INSTALL.rst.in]: Please two newline before headings-

File: INSTALL.rst.in [hunk 2]
```diff
@@ -175,3 +175,36 @@ Please see the detailed instructions for Windows in the file
...
+Bash on Windows
+---------------
+Using the latest insider preview of Windows adds a subsystem for Linux and users can use Bash just like a Linux shell. In order to set-up RtMidi,
```
Comment (SpotlightKid) [inline INSTALL.rst.in]: Please a newline after a heading.

File: INSTALL.rst.in [hunk 3]
```diff
@@ -175,3 +175,36 @@ Please see the detailed instructions for Windows in the file
...
+there are a few additional setup steps.
+
+* Setup the Pip and Python
+    * `apt-get remove upstart`
+    * `apt-get remove udev`
+    * `apt-get autoremove`
+    * `sudo apt-get install python-pip`
+    * `sudo apt-get install python-dev`
+    * `nano /usr/sbin/policy-rc.d`
+    * Write in this file the following:
```
Comment (SpotlightKid) [inline INSTALL.rst.in]: Double colon?

File: INSTALL.rst.in [hunk 4]
```diff
@@ -175,3 +175,36 @@ Please see the detailed instructions for Windows in the file
...
 .. _pyliblo: http://das.nasophon.de/pyliblo/
 .. _pyyaml: https://pypi.python.org/pypi/PyYAML
+
+Bash on Windows
+---------------
+Using the latest insider preview of Windows adds a subsystem for Linux and users can use Bash just like a Linux shell. In order to set-up RtMidi,
+there are a few additional setup steps.
+
+* Setup the Pip and Python
+    * `apt-get remove upstart`
```
Comment (SpotlightKid) [inline INSTALL.rst.in]: Double backticks for inline code, please. The file is in ReST syntax, not markdown.

**Comment by SpotlightKid** [comment]
Can you please put the new section in the file `INSTALL-windows.rst` and refer to it from this file?

**Comment by Fleker** [comment]
Okay, I made those changes.

**Comment by SpotlightKid** [comment]
Ok, I'll try to review and merge it tomorrow.

**Comment by SpotlightKid** [comment]
I'm sorry, but I think I'm not going to merge this.

Apart from that the text and formatting still contains some errors* and some of the commands listed require IMHO further explanation of why they are needed, I think that the Linux subsystem on Windows is still too new and it's only available as preview anyway. Things are likely to change rapidly for a while and errors will get fixed and workaround, like the ones described here might soon not be necessary anymore.

I therefor think that these instructions are best placed in the wiki under [User contributed documentation](../wiki/User-contributed-documentation) for now. When things stabilize, they can be incorporated in the "official" docs later.

I'll insert a reference to the wiki docs into the readme and `INSTALL*` files.

\* See the diff below for things I would change. There are also issues with mixed CRLF/LF line endings and no-newline-at-end-of-file, which I excluded from the diff.

```
diff --git a/INSTALL-windows.rst b/INSTALL-windows.rst
index 3699040..053c05f 100644
--- a/INSTALL-windows.rst
+++ b/INSTALL-windows.rst
@@ -130,41 +130,49 @@ Compiling with MinGW also does not work out-of-the-box yet. If you have any
 useful hints, please let the author know.
 
 
-Install through Bash on Windows
--------------------------------
+Installing python-rtmidi through Bash on Windows
+================================================
 
-Using the latest insider preview of Windows adds a subsystem for Linux and users can use Bash just like a Linux shell. In order to set-up RtMidi,
-there are a few additional setup steps.
+The latest insider preview of Windows 10 adds a subsystem for Linux and users
+can use Bash just like a Linux shell. In order to set up python-rtmidi, there
+are a few additional required steps before the normal installation procedure:
 
-- Setup the Pip and Python
-    - ``apt-get remove upstart``
-    - ``apt-get remove udev``
-    - ``apt-get autoremove``
-    - ``sudo apt-get install python-pip``
-    - ``sudo apt-get install python-dev``
-    - ``nano /usr/sbin/policy-rc.d``
-    - Write in this file the following:
+- Set up pip and Python (``$`` denotes the bash prompt)::
 
-    ::
+        $ apt-get remove upstart
+        $ apt-get remove udev
+        $ apt-get autoremove
+        $ sudo apt-get install python-pip
+        $ sudo apt-get install python-dev
+        $ nano /usr/sbin/policy-rc.d
+
+- Write the following to ``/usr/sbin/policy-rc.d``::
 
         #!/bin/sh
         exit 101
 
-    - ``chmod +x /usr/sbin/policy-rc.d``
-    - ``dpkg-divert --local --rename --add /sbin/initctl``
-    - ``ln -s /bin/true /sbin/initctl``            
-- Setup RtMidi
-    - ``sudo apt-get download udev``
-    - ``sudo apt-get install initramfs-tools`` 
-    - ``sudo dpkg -i --force-all confmiss udev_*.deb``
-    - ``apt-get install -y alsa-utils``
-    - ``gpasswd -a root audio``
-
-Unfortunately, there are some issues around the current version of Bash on Windows which prevents the Alsa and soundcards from being accessible.
-(Try running `alsamixer` and `aplay -l` to see if it works on your device)
-An issue_ about this has been filed. 
-
-Still, RtMidi can be installed and is technically accessible through Python.
+  Then::
+
+        $ chmod +x /usr/sbin/policy-rc.d
+        $ dpkg-divert --local --rename --add /sbin/initctl
+        $ ln -s /bin/true /sbin/initctl
+
+- Set up requirements for python-rtmidi::
+
+        $ sudo apt-get download udev
+        $ sudo apt-get install initramfs-tools
+        $ sudo dpkg -i --force-all confmiss udev_*.deb
+        $ apt-get install -y alsa-utils
+        $ gpasswd -a root audio
+
+Unfortunately, there are some issues around the current version of Bash on
+Windows, which prevents the Alsa and soundcards from being accessible. (Try
+running ``alsamixer`` and ``aplay -l`` to see if it works on your device). An
+issue_ about this has been filed.
+
+Still, after performing the above setup, python-rtmidi can be installed as
+described in the main ``INSTALL.rst`` file and is technically accessible
+through Python.
 
 
 .. _ez_setup.py: https://bitbucket.org/pypa/setuptools/raw/bootstrap/ez_setup.py
```

**Comment by Fleker** [comment]
That's fair. It seems like some of the issues stem from a lack of hardware support in the Linux subsystem: https://github.com/Microsoft/BashOnWindows/issues/242

**Pull request closed without merging**